engine/src/main/java/org/pentaho/di/kitchen/KitchenCommandExecutor.java

```diff
@@ -298,13 +300,31 @@ public Job loadJobFromRepository( Repository repository, String dirName, String
     return new Job( repository, jobMeta );
   }
 
-  public Job loadJobFromFilesystem( final String initialDir, final String filename ) throws Exception {
+  public Job loadJobFromFilesystem( String initialDir, String filename, String base64Zip ) throws Exception {
 
     if ( Utils.isEmpty( filename ) ) {
       System.out.println( BaseMessages.getString( getPkgClazz(), "Kitchen.Error.canNotLoadJob" ) );
       return null;
     }
 
+    if ( !Utils.isEmpty( base64Zip ) ) {
+      //expected form of filename "*.kjb"
+      String zipPath = Const.getUserHomeDirectory() + File.separator + java.util.UUID.randomUUID().toString() + ".zip";
+      filename = "zip:file:" + File.separator + File.separator + zipPath + "!" + filename;
+      File zipFile;
+
+      //responsibly attempt to write to file
+      try {
+        zipFile = decodeBase64StringToFile( base64Zip, zipPath );
+      } catch ( IOException e ) {
+        getLog().logError( e.toString() + "\n" );
+        e.printStackTrace();
```

Comment:
Do we need to log the stackTrace .. we are loggin the execption string a line before